src/plugins/intel_gpu/src/graph/dft.cpp

```diff
@@ -14,22 +14,27 @@ primitive_type_id dft::type_id() {
     return &instance;
 }
 
-layout typed_primitive_inst<dft>::calc_output_layout(const dft_node& node) {
+layout dft_inst::calc_output_layout(const dft_node& node) {
     auto primitive = node.get_primitive();
-    auto input_layout = node.input().get_output_layout();
+    const auto input_layout = node.input().get_output_layout();
 
     std::vector<tensor::value_type> dims_converted(primitive->output_shape.begin(), primitive->output_shape.end());
-    auto output_format = input_layout.format;
+
+    // Extend output layout for IRDFT case to make output rank match input rank
+    if (primitive->kind == dft_kind::inverse && primitive->mode == dft_mode::real) {
+        dims_converted.push_back(1);
+    }
 
     // Extend shape to 4d by pushing ones before the last dim
     for (auto i = dims_converted.size(); i < 4; ++i) {
         dims_converted.insert(std::prev(dims_converted.end()), 1);
     }
 
+    const auto output_format = format::get_preserved_blocked_format(dims_converted.size(), input_layout.format);
```

Comment:
No, for (I)DFT it should not be preserved, but we handle this inside this function, so not sure whether we need to add one more if here. Or maybe if we will add if, it will be more obvious.

Reply:
Yeah, by this if - https://github.com/openvinotoolkit/openvino/pull/12010/files#diff-2284fa8fd15e180af2b77e4fb16fa1f5a319bbf23c996e98ee30141042f472e5R330
But I think it's common enough, no?